### PR TITLE
IBX-7276: New User / Field twig functions

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | [IBX-7276](https://issues.ibexa.co/browse/IBX-7276) |
| Versions      | master |
| Edition       | all |

Document new Twig functions added as part of [301](https://github.com/ibexa/core/pull/301)

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date

### CODE REVIEW

File: docs/templating/twig_function_reference/field_twig_functions.md [hunk 1]
```diff
@@ -175,3 +178,31 @@ For example, use `ibexa_field_is_empty()` to check whether a Field is empty or f
...
 {% endif %}
 ```
+
+### `ibexa_field_group_name()`
+
+`ibexa_field_group_name()` returns a human-readable name of a Field group.
+
+| Argument | Type | Description |
+|---------------|------|-------------|
+| `content` | `Ibexa\Contracts\Core\Repository\Values\Content\Content` | Content item the Field group belongs to. |
```
Comment (adamwojs) [inline docs/templating/twig_function_reference/field_twig_functions.md:188]: ibexa_field_group_name except field group identifier, and not content item.

File: docs/templating/twig_function_reference/user_twig_functions.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+``` html+twig
+{{ ibexa_user_get_current().login }}
+```
+
+You can get the underlying Content item, for example the user's name,
+by accessing the `content` property:
+
+``` html+twig
+{{ ibexa_render(ibexa_user_get_current().content) }}
+```
```
Comment (adamwojs) [inline docs/templating/twig_function_reference/user_twig_functions.md:21]: This example is incorrect.

**Review by adamwojs — CHANGES_REQUESTED** [review]

**Comment by adamwojs** [comment]
Site note: ibexa_get_current_user has been renamed to ibexa_current_user

File: docs/templating/twig_function_reference/user_twig_functions.md (hunk 2 above)
Comment (dabrt) [reply docs/templating/twig_function_reference/user_twig_functions.md:21]: @adamwojs 
The example was merely moved from a different place in the docs.
Please look [here](https://doc.ibexa.co/en/latest/templating/twig_function_reference/other_twig_filters/). 
If the original example was wrong, please help me with a new one.

BTW, what is the difference between `ibexa_user_get_current()` and `ibexa_get_current_user()`?

Comment (adriendupuis) [reply docs/templating/twig_function_reference/user_twig_functions.md:21]: The actual example is risky. On a clean install, while using Anonymous, it ends with this error:
`An exception has been thrown during the rendering of a template ("The User does not have the 'read|view_embed' 'content' permission with: contentId '10', locationId 'n/a'").` (`ibexa_render` default view type is [`embed`](https://github.com/ibexa/core/blob/v4.6.0-beta3/src/lib/MVC/Symfony/Templating/RenderContentStrategy.php#L21)).

The `embed` view is a link to the Content item, the link's text is the Content item's name. So, it does a bit more than the example introduction says.

To just display the name, `{{ ibexa_content_name(ibexa_user_get_current()) }}` is enough and can be used with Anonymous.

But to stick to the introduction, I would propose an example not displaying anything but accessing as promised:
`{% set current_user_name = ibexa_user_get_current().content.name %}`

Comment (adriendupuis) [reply docs/templating/twig_function_reference/user_twig_functions.md:21]: `{{ ibexa_render_field(ibexa_user_get_current().content, 'last_name') }}` looks good to me.

Comment (adriendupuis) [reply docs/templating/twig_function_reference/user_twig_functions.md:21]: `{{ dump(ibexa_user_get_current().fields, ibexa_user_get_current().content.fields) }}` shows a difference between the two. But I do not find an example which would stay on a recommandable path…

Maybe we could avoid providing a working but ugly example and just say:

```markdown
You can get the underlying Content item by accessing the `content` property: `ibexa_user_get_current().content`
```

Or consider that specifying `.content` is a best practice over letting some automatic casting to happen:

---

You can get the underlying Content item, for example to display the user's last name,
by accessing the `content` property:

``` html+twig
{{ ibexa_render_field(ibexa_user_get_current().content, 'last_name') }}
```

---

File: docs/templating/twig_function_reference/user_twig_functions.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+``` html+twig
+{{ ibexa_user_get_current().login }}
+```
+
+You can get the underlying Content item, for example the user's name,
+by accessing the `content` property:
+
+``` html+twig
+{{ ibexa_render(ibexa_user_get_current().content) }}
```
Comment (adriendupuis) [inline docs/templating/twig_function_reference/user_twig_functions.md:20]: {{ ibexa_render_field(ibexa_user_get_current().content, 'last_name') }}

File: docs/templating/twig_function_reference/user_twig_functions.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+### `ibexa_user_get_current()`
+
+`ibexa_user_get_current()` returns the User object (`Ibexa\Contracts\Core\Repository\Values\User\User`) of the current user.
+
+``` html+twig
+{{ ibexa_user_get_current().login }}
+```
+
+You can get the underlying Content item, for example the user's name,
+by accessing the `content` property:
```
Comment (adriendupuis) [inline docs/templating/twig_function_reference/user_twig_functions.md:17]:
```suggestion
You can get the underlying Content item, for example to display the user's last name,
by accessing the `content` property:
```

File: docs/templating/twig_function_reference/user_twig_functions.md (hunk 3 above)
Comment (adriendupuis) [inline docs/templating/twig_function_reference/user_twig_functions.md:20]:
```suggestion
{{ ibexa_render_field(ibexa_user_get_current().content, 'last_name') }}
```

File: docs/templating/twig_function_reference/user_twig_functions.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+### `ibexa_current_user`
+
+The `ibexa_current_user()` function returns the User object (`Ibexa\Contracts\Core\Repository\Values\User\User`) of the current repository user.
+
+#### Examples
+
+``` html+twig
+Current user: {{ ibexa_current_user().login }}
+```
```
Comment (adriendupuis) [inline docs/templating/twig_function_reference/user_twig_functions.md:31]: https://github.com/ibexa/user/blob/main/src/bundle/Twig/UserExtension.php#L23-L24
```suggestion
### `ibexa_current_user()`

`ibexa_current_user()` is a deprecated alias of `ibexa_user_get_current()`.

```

File: docs/templating/twig_function_reference/user_twig_functions.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+### `ibexa_is_current_user`
```
Comment (adriendupuis) [inline docs/templating/twig_function_reference/user_twig_functions.md:33]:
```suggestion
### `ibexa_is_current_user()`
```

**Review by adriendupuis — APPROVED** [review]

**Review by julitafalcondusza — APPROVED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**